src/target/riscv/riscv.c

```diff
@@ -3814,7 +3788,7 @@ static int riscv_openocd_step_impl(struct target *target, int current,
 		}
 
 _exit:
-	if (enable_triggers(target, trigger_state) != ERROR_OK) {
+	if (enable_watchpoints(target, wps_to_enable) != ERROR_OK) {
 		success = false;
 		LOG_TARGET_ERROR(target, "Unable to enable triggers.");
```

Comment:
If you like: I have tried to make the error message even more clear:

```suggestion
		LOG_TARGET_ERROR(target, "Failed to re-enable watchpoint triggers after single-step.");
```

Reply:
Addressed.